apps/astarte_trigger_engine/lib/astarte_trigger_engine/events_consumer.ex

```diff
@@ -185,6 +190,28 @@ defmodule Astarte.TriggerEngine.EventsConsumer do
     []
   end
 
+  # If introspection = nil, it means we are using the introspection map format
+  # instead of the old one (pre-1.2) where introspection is a string
+  defp maybe_normalize_introspection_event(
+         %IncomingIntrospectionEvent{introspection: nil} = event
+       ) do
+    %IncomingIntrospectionEvent{introspection_map: introspection_map} = event
+
+    Enum.reduce(introspection_map, fn {interface_name, version}, acc ->
+      %InterfaceVersion{
+        major: version_major,
+        minor: version_minor
+      } = version
+
+      Map.put_new(acc, interface_name, %{
+        major: version_major,
+        minor: version_minor
+      })
+    end)
+  end
+
+  defp maybe_normalize_introspection_event(event), do: event
```

Comment:
Why isn't this behind the feature flag env? If we maintain compatibility for AMQP triggers I though we were going to maintain it also for HTTP triggers

Reply:
Compatibility is maintained: as the IncomingIntrospectionEvent [protobuf](https://github.com/astarte-platform/astarte_core/blob/5335b0d6e9cd33c302a69c3ac905787515244750/lib/astarte_core/triggers/simple_events/incoming_introspection_event.proto#L27) has `introspection` set to `nil` only when `introspection_map` is not nil (see the [relevant part of this PR on DUP](https://github.com/astarte-platform/astarte/pull/883/files#diff-e4f382e29d9d340ecc3a4c68339fc94273df0e60f41e8e07c80c55e414f06aeaR175-R180)), an event with the old format will not match the first head and fall back safely into the second, being left as it is (just an introspection string)